common.go

```diff
@@ -77,4 +77,18 @@ type WebView interface {
 	// f must be a function
 	// f must return either value and error or just error
 	Bind(name string, f interface{}) error
+
+	// NavigationStarting binds a callback function for right before the webview2
+	// perform a navigation
+	//
+	// f must be a function
+	// f must return true, to proceed with the navigation
+	// f must return false, to cancel the navigation
+	NavigationStarting(f func(additionalAllowedFrameAncestors string, isRedirected bool, isUserInitiated bool, navigationId uint64, uri string) bool) error
```

Comment:
As mentioned elsewhere, the `WebView` interface is frozen to preserve compatibility with `webview/webview`. The best way to add new functionality like this right now is to allow the user to pass a handler into the options struct of `NewWithOptions`.